src/components/layouts/articleAdministration/addNpkProductDialog.js

```diff
@@ -0,0 +1,155 @@
+import React, {useEffect, useState} from 'react';
+import DynamicDialog from "../../dynamicDialog/dynamicDialog";
+import PropTypes, {func} from "prop-types";
+import TextField from "@material-ui/core/TextField";
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import {getNpks, submitNewProduct} from "../../connectionHandler/connectionHandler";
+import MenuItem from "@material-ui/core/MenuItem";
+
+const AddNpkProductDialog = ({setErrorMessage, onCancel, onSubmit, show, setProducts, npks, setNpks, ...props}) => {
+
+  const cancelButtonText = 'Abbrechen';
+  const acceptButtonText = 'Bestätigen';
+  const title = 'Neuen NPK Artikel erstellen';
+  const text = 'Füllen Sie alle Felder ab um einen neuen NPK Artikel zu erstellen.';
+  const textfields = [
+    {
+      id: 'number',
+      label: 'Artikelnummer',
+      type: 'textarea',
+      required: true
+    },
+    {
+      id: 'price',
+      label: 'Preis',
+      type: 'number',
+      required: true
+    },
+    {
+      id: 'unit',
+      label: 'Einheit',
+      type: 'textarea',
+      required: true,
+      select: true,
+      options: [
+        {
+          name: 't',
+          value: 'tons'
+        },
+        {
+          name: 'Stk.',
+          value: 'pieces'
+        },
+        {
+          name: 'm',
+          value: 'meter'
+        },
+        {
+          name: 'kg',
+          value: 'kilo'
+        },
+        {
+          name: 'l',
+          value: 'liter'
+        },
+        {
+          name: 'qm',
+          value: 'squarem'
+        },
+        {
+          name: 'h',
+          value: 'hours'
+        },
+      ],
+    },
+    {
+      id: 'description',
+      label: 'Beschreibung',
+      type: 'textarea',
+      required: true
+    }
+  ];
+
+  const saveNewArticle = (articleData) => {
+    submitNewProduct(articleData, setErrorMessage, onSubmit);
+  }
+
+  const parseArticleData = (articleData) => {
+    articleData.price = parseInt(articleData.price);
+    articleData.name = (articleData.npk.split(" "))[1]
+    articleData.npk = (articleData.npk.split(" "))[0]
+    saveNewArticle(articleData);
+  }
+
+  useEffect(() => {
+    getNpks(setErrorMessage, setNpks)
```

Comment:
Are the npks needed outside of this dialog? If not, then the state for the npks shouldn't be outisde of the dialog, but within it.

Reply:
moved it out into the upper component